### PR TITLE
Vjhaveri/cache offset

### PR DESCRIPTION
**Pull request by vedantjh2**
## Motivation

`run_lora_b_sgemm` and `run_gate_up_lora` in the torch_native LoRA backend allocate new `torch.tensor` offset arrays on every call — once per layer, per batch. These tensors are small constant integer arrays (2-3 elements) used as slice boundaries, and their values are determined entirely by the weight dimensions which are fixed per layer. Allocating and discarding them repeatedly is unnecessary overhead.

## Modifications

Cache these offset tensors in dictionaries keyed by their dimensions. The first call for a given dimension creates the tensor; subsequent calls return the cached one. These tensors are never modified by consumers, so reuse is safe.

## Accuracy Tests

Used in house model and adapter. compared before and after the change:

```
jobuser [ ~ ]$ python test_pcg_accuracy.py --compare --file1 embeddings_current.json --file2 embeddings_baseline.json
Comparing current (from embeddings_current.json) vs baseline (from embeddings_baseline.json)
  current: 120 base, 120 lora
  baseline: 120 base, 120 lora

============================================================
  BASE MODEL: current vs baseline  (120 inputs)
============================================================
  Cosine similarity:
    min:  1.00000000
    mean: 1.00000000
    max:  1.00000000
  Max abs diff per embedding:
    min:  0.00000000e+00
    mean: 0.00000000e+00
    max:  0.00000000e+00
  Mean abs diff per embedding:
    min:  0.00000000e+00
    mean: 0.00000000e+00
    max:  0.00000000e+00

  VERDICT: PASS - outputs match

============================================================
  LORA MODEL: current vs baseline  (120 inputs)
============================================================
  Cosine similarity:
    min:  1.00000000
    mean: 1.00000000
    max:  1.00000000
  Max abs diff per embedding:
    min:  0.00000000e+00
    mean: 0.00000000e+00
    max:  0.00000000e+00
  Mean abs diff per embedding:
    min:  0.00000000e+00
    mean: 0.00000000e+00
    max:  0.00000000e+00

  VERDICT: PASS - outputs match

  current: base vs lora similarity = mean 0.350912, min 0.103902 (should NOT be 1.0 if LoRA is active)

  baseline: base vs lora similarity = mean 0.350912, min 0.103902 (should NOT be 1.0 if LoRA is active)
```

## Benchmarking and Profiling

Benchmarked on Qwen3-Embedding-0.6B with 4 LoRA adapters, 240 requests at rate 40 req/s (`--lora-backend torch_native`):

| Metric | Before | After | Delta |
|--------|--------|-------|-------|
| Request throughput (req/s) | 29.66 | 30.60 | **+3.2%** |
| Input token throughput (tok/s) | 171,502 | 176,945 | **+3.2%** |
| Mean E2E latency (ms) | 1413 | 1207 | **-14.6%** |
| Median E2E latency (ms) | 1510 | 1305 | **-13.6%** |
| P99 E2E latency (ms) | 2128 | 1804 | **-15.2%** |

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by hnyls2002** [comment]
@vedantjh2 I'm fairly sure this is already addressed upstream: PR #21885 reworked the torch_native LoRA backend so run_lora_b_sgemm and run_gate_up_lora now take a pre-computed output_offset_cpu argument (python/sglang/srt/lora/backend/torch_backend.py:83 and :134), so the per-call offset-tensor allocation this PR caches no longer happens in those functions. Closing as superseded; please reopen if I've missed something.
